docs/getting-started/setup.md

```diff
@@ -1,59 +1,239 @@
+  - **Recommended.** Faster, somewhat less convenient.
+- [Docker for Mac](#install-macos-docker-for-mac)
+  - Somewhat slower, but easier to use and update
+
+<a name="install-macos-virtualbox"></a>
+### macOS with VirtualBox
+
+With this method, Docker will run inside the VM in VirtualBox.
+
+1. Download and Install VirtualBox
+
+    [![Download VirtualBox 5.2.2](https://img.shields.io/badge/⇩%20-Virtual%20Box%205.2.2-green.svg)](http://download.virtualbox.org/virtualbox/5.2.2/VirtualBox-5.2.2-119230-OSX.dmg)
+
+1. Enable Kernel extension ([Why?](https://developer.apple.com/library/content/technotes/tn2459/_index.html))
+
+    Go to `System Preferences > Security & Privacy`.  
+    If you do not see the Allow button it means the extension is already enabled.
+
+    ![Allowing VirtualBox kernel extension](../_img/virtualbox-kernel-extension-allow.png)
+
+1. Open Terminal app and run
+
+    ```bash
+    curl -fsSL get.docksal.io | bash
+    ```
+
+<a name="install-macos-docker-for-mac"></a>
+### macOS with Docker for Mac
+
+1. Download and Install Docker for Mac
+
+    [![Docker for Mac](https://img.shields.io/badge/⇩%20-Docker%20For%20Mac-green.svg)](https://download.docker.com/mac/stable/Docker.dmg)
+
+1. Start Docker for Mac
+
+    Wait until it says "Docker is running" in the menubar icon menu.
+
+1. Open Terminal app and run
+
+    ```bash
+    curl -fsSL get.docksal.io | DOCKER_NATIVE=1 bash
+    ```
+
+<a name="install-linux"></a>
+### Linux installation options 
+
+Click your repo to proceed to docs.
+
+- [Ubuntu](#install-linux-debian-fedora) 
+- [Mint](#install-linux-debian-fedora) 
+- [Debian](#install-linux-debian-fedora) 
+- [Fedora](#install-linux-debian-fedora) 
+- [CentOS](#install-linux-debian-fedora) 
+- [Other distribution](#install-linux-other)
+
+<a name="install-linux-debian-fedora"></a>
+### Linux. Debian, Ubuntu, Fedora.
+
+Debian with all derivatives (Ubuntu, Raspbian etc.), Ubuntu with derivatives (Mint etc.), 
+and Fedora with derivatives are supported out of the box with automatic installation. 
+
+1. Check pre-requisites
+
+    By default, Apache listens on `0.0.0.0:80` and `0.0.0.0:443`. 
+    This will prevent Docksal reverse proxy from running properly. 
+    You can resolve it an any of the following ways:
+      - Reconfigure Apache to listen on different host (e.g., `127.0.0.1:80` and `127.0.0.1:443`)
+      - Reconfigure Apache to listen on different ports (e.g., `8080` and `4433`)
+      - Stop and disable Apache
+
+1. Check your software. 
+
+    Check that you have installed and configured
+    - curl
+    - sudo
+    
+1. Open Terminal and run
+
+    ```bash
+    curl -fsSL get.docksal.io | bash
+    ```
+
+<a name="install-linux-other"></a>
+### Linux. Other distributions
+
+#### Compatibility
+
+If you cannot find your distribution in the list above, it does not mean it is not supported! 
+Less known Debian, Ubuntu or Fedora derivatives are most likely supported.
```

Comment:
"Less known" is okay here, but "Lesser known" is better.
There should be comma after Ubuntu.